cranelift/codegen/src/isa/aarch64/abi.rs

```diff
@@ -180,31 +251,52 @@ fn compute_arg_locs(call_conv: isa::CallConv, params: &[ir::AbiParam]) -> (Vec<A
             ),
         }
 
-        if in_int_reg(param.value_type) {
-            if let Some(param) = try_fill_baldrdash_reg(call_conv, param) {
-                ret.push(param);
-            } else if next_xreg < 8 {
-                ret.push(ABIArg::Reg(xreg(next_xreg).to_real_reg(), param.value_type));
-                next_xreg += 1;
-            } else {
-                ret.push(ABIArg::Stack(next_stack as i64, param.value_type));
-                next_stack += 8;
-            }
-        } else if in_vec_reg(param.value_type) {
-            if next_vreg < 8 {
-                ret.push(ABIArg::Reg(vreg(next_vreg).to_real_reg(), param.value_type));
-                next_vreg += 1;
+        let intreg = in_int_reg(param.value_type);
+        let vecreg = in_vec_reg(param.value_type);
+        debug_assert!(intreg || vecreg);
+        debug_assert!(!(intreg && vecreg));
```

Comment:
I thought about that too :-) But it's a little obscure that it works for bools (at least my C-trained brain was unsure) so I think it might be better to be explicit...